### ISSUE TITLE
LInux Debian, super user authority

### ISSUE THREAD

**Issue by Calibos** [FTBTeam/FTB-App#1194]
### What Operating System

Linux (Debian)

### Debug Code

Debug tool didn't run

### Describe the bug

Sadly, I couldn't get your debug tool to run. When I start the launcher, it tries to run in Admin mode, pulls up a password dialogue box. If you do anything but allow it, it crashes, and begins cycling instances of itself ending, and restarting, several at a time. I can't close them in the system manager before they've stopped themselves and started another one, it cycles fast enough that there appear to be multiple instances running. The only way I can end the cycle is rebooting, or logging out. I'm not sure why the program needs admin mode, and I'm uncomfortable running it that way.

### Steps to reproduce

Start the FTB launcher, deny it access to admin mode.

### Expected behaviour

Start FTB app, program opens, launch minecraft instance

### Screenshots

![image](https://github.com/user-attachments/assets/a3aed71b-ef75-4e45-a885-4affe24c6fd1)

**Comment by klugemonkey** [comment]
IMHO, should probably allow disabling or disable altogether updates from within the client for .deb installs.  The client should probably inform the user that an update is available, but let root update it outside the client since that is how they installed it in the first place.

**Comment by MichaelHillcox** [comment]
IIRC, this is the only route for updating we have. Are you inferring users should only be informed of an update and told to source it themselves?

**Comment by MichaelHillcox** [comment]
> Sadly, I couldn't get your debug tool to run. When I start the launcher, it tries to run in Admin mode, pulls up a password dialogue box. If you do anything but allow it, it crashes, and begins cycling instances of itself ending, and restarting, several at a time. I can't close them in the system manager before they've stopped themselves and started another one, it cycles fast enough that there appear to be multiple instances running. The only way I can end the cycle is rebooting, or logging out. I'm not sure why the program needs admin mode, and I'm uncomfortable running it that way.

Yeah this is very weird. I've not seen this before myself... The app would not do anything bad in admin mode, you can see this from the code in this repo but I do agree it shouldn't need admin mode. I'm not 100% sure on how this can be fixed but I can look into it. It's likely a quirk with Electron

**Comment by klugemonkey** [comment]
> IIRC, this is the only route for updating we have. Are you inferring users should only be informed of an update and told to source it themselves?

Yes, that's exactly what I'm saying.  If an application is installed as a .deb and requires root to install, then you probably don't want to be provide a whole new means to install packages by prompting for root access on the user side, through an electon library to pkexec to update their .dpkg.   Inform the user that there is an update that requires root access to update.  Allow them to download the .deb and install with command line, since that's how they installed it in the first place.  

If they want a GUI for their installs, consider supporting apt and package managers by creating a repository that can be added with a sources.list.d .list file which points at the package repository, as well as letting them download the repository public key for installation.   This is how most packages are installed in linux.   Then at least they won't have to download it manually, but just check for updates and upgrade.

Last thing I want as an admin, is to have an Electron App use a third party API to call pkexec (PolicyKit aka PwnKit), to prompt me for my root credentials.

**Comment by klugemonkey** [comment]
But i think the main issue is that it's automatically trying to upgrade, not even let the user initiate on their own or decide to handle manually.   Nor can it be disabled.

**Comment by klugemonkey** [comment]
Changing it such that automatic updates would be an option switch, providing an upgrade button for those that want to handle from user,  showing in the gui that there is an update available would be preferable to forcing upgrade at start with nonstandard installation methods.
